### PR TITLE
fix: make wordGrades array in options readonly and make mutable in class

### PR DESCRIPTION
**Pull request by gcottrell1**
😅

### CODE REVIEW

File: lib/node/GradedText.ts [hunk 1]
```diff
@@ -13,7 +13,8 @@ export class Generic<IExtension> implements IMutable {
 
   constructor(init: Init<IExtension>) {
     const { wordGrades, isCorrect, score, copy, ...rest } = init;
-    this.wordGrades = wordGrades;
+    // tslint:disable-next-line:prefer-object-spread
+    this.wordGrades = wordGrades.map(wordGrade => Object.assign({}, wordGrade));
```
Comment (leethree) [inline lib/node/GradedText.ts]: what does this do?

Comment (gcottrell1) [reply lib/node/GradedText.ts]: makes word grades mutable as the incoming type is immutable

Comment (leethree) [reply lib/node/GradedText.ts]: i think immutability is just Typescript types. it's not enforced in runtime?

Comment (leethree) [reply lib/node/GradedText.ts]: try `wordGrade => {...wordGrade}`

**Pull request merged**